core/chains/evm/config/toml/defaults.go

```diff
@@ -33,48 +32,83 @@ var (
 )
 
 func init() {
-	// read the defaults first
+	// read all default configs
+	initReadDefaults()
+
+	// overrides should be applied as:
+	// fallback.toml(defaults dir) <- fallback.toml(env CL_CHAIN_FALLBACK) <- ChainSpecific.toml(env CL_CHAIN_DEFAULTS)
+	//
+	// the custom fallback gets processed and overrides the default fallback
+	if path := env.CustomFallback.Get(); path != "" {
+		_, chain, err := readConfig(path, os.ReadFile, true)
+		if err != nil {
+			if !errors.Is(err, errFallbackConfig) {
+				log.Fatalf("custom fallback config error: %s", err.Error())
+			}
+
+			fallback = chain
+		}
+	}
+
+	// check for and apply any overrides
+	initApplyEVMOverrides()
+}
 
+var (
+	errRead           = errors.New("error reading file")
+	errDecode         = errors.New("error in TOML decoding")
+	errMissingChainID = errors.New("missing ChainID")
+	errNonNilChainID  = errors.New("fallback ChainID must be nil")
+	errFallbackConfig = errors.New("fallback config")
+)
+
+func initReadDefaults() {
+	// read the defaults first
 	fes, err := defaultsFS.ReadDir("defaults")
 	if err != nil {
-		log.Fatalf("failed to read defaults/: %v", err)
+		log.Fatalf("failed to read defaults: %v", err)
 	}
+
 	for _, fe := range fes {
-		path := filepath.Join("defaults", fe.Name())
-		b, err2 := defaultsFS.ReadFile(path)
-		if err2 != nil {
-			log.Fatalf("failed to read %q: %v", path, err2)
+		if fe.IsDir() {
+			// Skip directories
+			continue
 		}
-		var config = struct {
-			ChainID *big.Big
-			Chain
-		}{}
 
-		if err3 := cconfig.DecodeTOML(bytes.NewReader(b), &config); err3 != nil {
-			log.Fatalf("failed to decode %q: %v", path, err3)
-		}
-		if fe.Name() == "fallback.toml" {
-			if config.ChainID != nil {
-				log.Fatalf("fallback ChainID must be nil, not: %s", config.ChainID)
+		// read the file to bytes
+		path := filepath.Join("defaults", fe.Name())
+		chainID, chain, err := readConfig(path, defaultsFS.ReadFile, fe.Name() == "fallback.toml")
+		if err != nil {
+			if errors.Is(err, errFallbackConfig) {
+				fallback = chain
+
+				continue
 			}
-			fallback = config.Chain
-			continue
-		}
-		if config.ChainID == nil {
-			log.Fatalf("missing ChainID: %s", path)
+
+			log.Fatal(err.Error())
 		}
-		DefaultIDs = append(DefaultIDs, config.ChainID)
-		id := config.ChainID.String()
+
+		// add ChainID to set of default IDs
+		DefaultIDs = append(DefaultIDs, chainID)
```

Comment:
I think it is important not to separate these global var assignments (`defaults` and `defaultNames` too) from their declarations.

Reply:
If you want to use helpers, then how about returning values to from `init()`?